conbench/util.py

```diff
@@ -19,6 +22,44 @@
 adapter = HTTPAdapter(max_retries=retry_strategy)
 
 
+log = logging.getLogger()
+
+
+def tznaive_iso8601_to_tzaware_dt(
+    input: Union[str, list[str]]
+) -> Union[datetime, list[datetime]]:
+    """
+    Convert time strings into datetime objects.
+
+    If a list of strings is provided return a list of datetime objects.
+
+    If a single string is provided return a single datetime object.
+
+    Assume that each provided string is in ISO 8601 notation without timezone
+    information, but that the time is actually meant to be interpreted in the
+    UTC timezone.
+
+    Note: this was built with and tested for a value like 2022-03-03T19:48:06
+    which in this example represents a commit timestamp (in UTC, additional
+    knowledge).
+    """
+
+    def _convert(s: str):
+        # Do some sanity-checking. If this happens, only emit the warning
+        # but then still hard-set UTC timezone.
+        if "Z" in s or "+" in s:
+            log.warning("expected tz-naive timestring, but saw: %s", s)
+
+        return datetime.fromisoformat(s).replace(tzinfo=timezone.utc)
```

Comment:
Incredibly important: the `replace()` on tzinfo is non-magic, i.e. it does not try to convert from one timezone into another by tweaking the numerical values of the hour, etc.

Corresponding docs:

> If you merely want to attach a time zone object tz to a datetime dt without adjustment of date and time data, use dt.replace(tzinfo=tz)

from https://docs.python.org/3/library/datetime.html.

(a common mistake is to use `datetime.astimezone(tz=None)` which tries to be smart, often too smart).